src/main/kotlin/com/github/mitchellolsthoorn/testgenie/evosuite/Runner.kt

```diff
@@ -2,132 +2,107 @@ package com.github.mitchellolsthoorn.testgenie.evosuite
 
 import com.github.mitchellolsthoorn.testgenie.settings.TestGenieSettingsService
 import com.intellij.execution.configurations.GeneralCommandLine
-import com.intellij.execution.process.*
+import com.intellij.execution.process.OSProcessHandler
+import com.intellij.execution.process.ProcessAdapter
+import com.intellij.execution.process.ProcessEvent
+import com.intellij.execution.process.ProcessHandler
 import com.intellij.openapi.diagnostic.Logger
 import com.intellij.openapi.util.Key
 import com.intellij.openapi.util.io.FileUtilRt
 import java.io.File
 import java.nio.charset.Charset
 
+/**
+ * A utility class that runs evosuite as a separate process in its various
+ * modes of operation.
+ *
+ * @param projectPath The root of the project we're testing, this sets the working dir of the evosuite process
+ * @param projectClassPath Path to the class path containing the compiled classes. This will change according to
+ * build system (e.g. Maven target/classes or Gradle build/classes)
+ * @param classFQN Fully qualified name of the class under test
+ */
+class Runner(private val projectPath: String, private val projectClassPath: String, private val classFQN: String) {
+    private val log = Logger.getInstance(this::class.java)
+
+    private val evoSuiteProcessTimeout: Long = 12000000 // TODO: Source from config
+    private val javaPath = "java" // TODO: Source from config
+    private val evosuiteVersion = "1.0.2" // TODO: Figure out a better way to source this
+
+    private val pluginsPath = System.getProperty("idea.plugins.path")
+    private var evoSuitePath = "$pluginsPath/TestGenie/lib/evosuite-$evosuiteVersion.jar"
+
+    private val ts = System.currentTimeMillis()
+    private val sep = File.separatorChar
+    private val testResultDirectory = "${FileUtilRt.getTempDirectory()}${sep}testGenieResults$sep"
+    private val testResultName = "test_gen_result_$ts"
+    private val serializeResultPath = "\"$testResultDirectory$testResultName\""
+
+    private val settingsState = TestGenieSettingsService.getInstance().state
+
+    private var command = mutableListOf<String>()
+
+    /**
+     * Sets up evosuite to run for a target class. This is the simplest configuration.
+     */
+    fun forClass(): Runner {
```

Comment:
@return in KDoc is missing. Could you please add it?

Reply:
The return value here is just the same instance, I want to avoid cluttering documentation.